_posts/2020-12-15-why-developer-should-know-owasp.md

```diff
@@ -0,0 +1,71 @@
+
+# All software projects are also information security projects
+
+As we can see, information security must be considered in software projects of any size. Also, information security is something you can't just add to the top of the software afterward. It must be taken into account from the beginning, already in the planning phase. The baseline of information security should be something you can just rely on. Just as you expect to get the software of good quality, you should get secure software.
+
+Of course, there are software projects where the baseline of the information security must be higher, and achieving that level costs more money. However, I like to think that securely implementing basic things does not increase costs. If you decide to do, for example, SQL queries in a way where SQL injection is impossible from the beginning, it takes as much time as making them less securely. The costs increase only if you first make them less securely, and then in a more secure way afterward.
+
+![Bobby Tables we call him](/img/why-owasp/exploits_of_a_mom.png)
+
+*Bobby Tables we call him, a classic from [https://xkcd.com/327/](https://xkcd.com/327/)*
+
+In a software project, the possible risks must be examined carefully, and sufficient security level needs to be decided. If the information system is going to be used in a restricted network by a certain number of users, you might not need the same amount of security as in open banking systems. However, it is a decision which needs to be done knowingly. 
+
+# Where the information security comes from?
+
+The information security in a project comes from people. Depending on the project there might be some security specialists but all the people involved in the project need to aware of security issues.
+
+One security aspect is the practises used in the project. Do people behave securely? When working on a project, you can't tell everyone about it, at least not at a detailed level. You must not keep your passwords on post-its, you should use always good passwords, perhaps utilize a password manager, and enable multi-factor authentication whenever it's possible. Do not push sensitive data, such as database passwords, to version control. Use a VPN instead of unprotected WLANs. Recognize if the risk is bearable before pasting project-specific things into third-party services, like Pastebin, XML validator or some language translator. After all that the other aspect is being aware of different security threats which exist in a digital environment. 

```

Comment:
Perhaps: "at least not **on** a detailed level."

Reply:
fixed